pkg/meta/sql.go

```diff
@@ -2479,10 +2479,15 @@ func (m *dbMeta) doSyncDirStat(ctx Context, ino Ino) (*dirStat, syscall.Errno) {
 		if !exist {
 			return syscall.ENOENT
 		}
-		_, err = s.Insert(&dirStats{ino, stat.length, stat.space, stat.inodes})
-		if err != nil && isDuplicateEntryErr(err) {
-			// other client synced
-			err = nil
+		exist, err = s.Exist(&dirStats{Inode: ino})
```

Comment:
try `insert` first, and do update if it returns 'duplicated entry error'

Reply:
done